L/LibCURL/common.jl

```diff
@@ -0,0 +1,107 @@
+    cd $WORKSPACE/srcdir/curl-*
+
+    # Holy crow we really configure the bitlets out of this thing
+    FLAGS=(
+        # Disable....almost everything
+        --without-ssl --without-gnutls --without-gssapi
+        --without-libidn --without-libidn2 --without-librtmp
+        --without-nss --without-polarssl
+        --without-spnego --without-libpsl --disable-ares --disable-manual
+        --disable-ldap --disable-ldaps --without-zsh-functions-dir
+        --disable-static --without-libgsasl
+
+        # A few things we actually enable
+        --with-libssh2=${prefix} --with-zlib=${prefix} --with-nghttp2=${prefix}
+        --enable-versioned-symbols
+    )
+
+
+    if [[ ${target} == *mingw* ]]; then
+        # We need to tell it where to find libssh2 on windows
+        FLAGS+=(LDFLAGS="${LDFLAGS} -L${prefix}/bin")
+
+        # We also need to tell it to link against schannel (native TLS library)
+        FLAGS+=(--with-schannel)
+    elif [[ ${target} == *darwin* ]]; then
+        # On Darwin, we need to use SecureTransport (native TLS library)
+        FLAGS+=(--with-secure-transport)
+
+        # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+        # complain about: `Symbol not found: ___isOSVersionAtLeast`
+        if [[ "${target}" == aarch64* ]]; then
+            export CFLAGS=-mmacosx-version-min=11.0
+        else
+            export CFLAGS=-mmacosx-version-min=10.11
+        fi
+    else
+        # On all other systems, we use MbedTLS
+        FLAGS+=(--with-mbedtls=${prefix})
+    fi
+
+    ./configure --prefix=$prefix --host=$target --build=${MACHTYPE} "${FLAGS[@]}"
+    make -j${nproc}
+    make install
+    install_license COPYING
+    """
+
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = supported_platforms(;experimental=true)
+
+    # The products that we will ensure are always built
+    if this_is_curl_jll
+        # The Curl_jll package only provides the executable
+        products = [
+            ExecutableProduct("curl", :curl),
+        ]
+    else
+        # The LibCURL_jll *package* provides both the library and the executable.
+        # However, the LibCURL_jll *stdlib* only provides the library.
+        products = [
+            LibraryProduct("libcurl", :libcurl),
+            ExecutableProduct("curl", :curl),
+        ]
+    end
+
+    # Dependencies that must be installed before this package can be built
+    dependencies = [
+        Dependency("LibSSH2_jll"),
+        Dependency("Zlib_jll"),
+        Dependency("nghttp2_jll"),
+        # Note that while we unconditionally list MbedTLS as a dependency,
+        # we default to schannel/SecureTransport on Windows/MacOS.
+        Dependency("MbedTLS_jll"; compat="~2.28.0"),
+    ]
+
+    if this_is_curl_jll
+        # Curl_jll depends on LibCURL_jll
+        push!(dependencies, Dependency("LibCURL_jll"))
```

Comment:
It should probably depend on the same version?
```suggestion
        push!(dependencies, Dependency("LibCURL_jll"; compat="$(version)"))
```
I don't know what happens if you try to use a 7.81.0 curl with a 7.71.0 libcurl, but I don't expect anything good.

Reply:
I can't imagine that anything good would happen 😂